### ISSUE TITLE
Folders created within a list appear with their ID instead of their title when reloading list views

### ISSUE THREAD

**Issue by reneeichert1** [pnp/pnpjs#1413]
### Category
- [ ] Enhancement
- [x] Bug
- [x] Question
- [ ] Documentation gap/issue

### Version

Please specify what version of the library you are using: [2.0.10]

Please specify what version(s) of SharePoint you are targeting: [SharePoint Online]

### Expected / Desired Behavior / Question
I am using pnpjs to create a folder inside a list using the given code:
```
            let list = await sp.web.lists.getByTitle(this.config.listName);
            list.items.add({
                Title : folderName,
                ContentTypeId:'0x0120',
                FileSystemObjectType: 1
            }).then(({item}) => {
                return item.update({
                    Title: folderName,
                    FileLeafRef: this.config.listUrl + "/"+ folderName
                });
            });
```` 

The folder is created and shows up in the list with the desired name and fileref.

The folder should be accessible using a simple query and crawling the items with using an odata query with a filter and substringof method.

### Observed Behavior
The folder is only accessible when using the id of the list item of the folder followed by "_.000"
I. e.
creating a folder named "testfolder" inside as the first item of a list should be queryable with:
sp.web.lists.getByTitle('listname').items.filter("substringof('testfolder', FileDirRef)")
this does not return any results. But if the query is changed to:
sp.web.lists.getByTitle('listname').items.filter("substringof('1_.000', FileDirRef)") The query returns the items inside the folder (as expected)

### Steps to Reproduce

- Create a folder using the code above
- Have a look at the folder in the list
- Reload the list view inside the folder and have a look at the breadcrumb
- Query list items inside the folder using the code from above

![image](https://user-images.githubusercontent.com/40966739/96451123-68feaa00-1217-11eb-9d82-922731f284b6.png)

**Comment by patrick-rodgers** [comment]
Hmmm, that isn't a pattern I have used before or tested. Have you looked at the [folders module](https://pnp.github.io/pnpjs/sp/folders/#add)? It has dedicated methods for working with folders which is how I've always done it. Any reason you can't use that? Blocked by something?

**Comment by reneeichert1** [comment]
> Hmmm, that isn't a pattern I have used before or tested. Have you looked at the [folders module](https://pnp.github.io/pnpjs/sp/folders/#add)? It has dedicated methods for working with folders which is how I've always done it. Any reason you can't use that? Blocked by something?

Yes,

if you use the folders method to create folders inside lists, they won't appear in the view (they are basically invisible) but are queryable with REST. We need both functionalities :-)

**Comment by reneeichert1** [comment]
Oh, I almost forgot, the pattern comes from this issue here: #100

**Comment by koltyakov** [comment]
Hey @reneeichert1, 

Not a pattern I would say but a workaround for some older SharePoint versions.
In SPO, a folder item can be now created in a single call using new AddSubFolderUsingPath method:

```typescript
await sp.web.getFolderByServerRelativePath('/sites/site/Lists/AList')
    .addSubFolderUsingPath('Readable Name');
```

**Comment by patrick-rodgers** [comment]
@reneeichert1 - were you able to try out @koltyakov's suggestion? Working?

**Comment by reneeichert1** [comment]
@patrick-rodgers yes the suggestion is working as expected, sorry for the inconvenience.

I do have another question though: Adding a folder is only the first step, I cannot find any documentation on how to add list items to a subfolder created inside a custom list (would be a new issue if needed).
